pkg/infrastructure/azure/loadbalancer.go

```diff
@@ -65,13 +65,46 @@ func createPublicIP(ctx context.Context, in *lbInput) (*armnetwork.PublicIPAddre
 	return &resp.PublicIPAddress, nil
 }
 
-func createExternalLoadBalancer(ctx context.Context, pip *armnetwork.PublicIPAddress, in *lbInput) (*armnetwork.LoadBalancer, error) {
+func updateExternalLoadBalancer(ctx context.Context, pip *armnetwork.PublicIPAddress, in *lbInput) (*armnetwork.LoadBalancer, error) {
 	loadBalancerName := in.infraID
 	probeName := "api-probe"
 	frontEndIPConfigName := "public-lb-ip-v4"
 	backEndAddressPoolName := in.infraID
 	idPrefix := fmt.Sprintf("subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/loadBalancers", in.subscriptionID, in.resourceGroup)
 
+	// Get the CAPI-created outbound load balancer so we can modify it.
+	extLB, err := in.lbClient.Get(ctx, in.resourceGroup, in.infraID, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get external load balancer: %w", err)
+	}
+
+	// Get the existing frontend configuration and backend address pool and
+	// create an additional frontend configuration mand backend address
+	// pool. Use the newly created public IP address with the additional
+	// configuration so we can setup load balancing rules for the external
+	// API server.
+	frontendIPConfigs := extLB.Properties.FrontendIPConfigurations
+	if len(frontendIPConfigs) == 0 {
+		return nil, fmt.Errorf("could not get frontendIPConfigs for external LB %s", *extLB.Name)
+	}
+	additionalFrontendIPConfig := &armnetwork.FrontendIPConfiguration{
+		Name: &frontEndIPConfigName,
+		Properties: &armnetwork.FrontendIPConfigurationPropertiesFormat{
+			PrivateIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodDynamic),
+			PublicIPAddress:           pip,
+		},
+	}
+	frontendIPConfigs = append(frontendIPConfigs, additionalFrontendIPConfig)
+
+	backendAddressPools := extLB.Properties.BackendAddressPools
+	if len(backendAddressPools) == 0 {
+		return nil, fmt.Errorf("could not get backendAddressPools for external LB %s", *extLB.Name)
```

Comment:
Afaik, we are not retying this operation (updateExternalLoadBalancer). So, if we return from here, we would have added a frontendIPConfig but no backendAddressPools. 

We could either 1. Add the new additionalBackendAddressPool even if len(backendAddressPools) == 0 OR 2. Add additionalFrontendIPConfig and additionalBackendAddressPool together after all error checking is complete.